test/lifecycle/TimeFrame/TimeFrame.test.js

```diff
@@ -0,0 +1,154 @@
+const { time, shouldFail,BN } = require('openzeppelin-test-helpers');
+const { expectBNIsWithinDelta } = require('../../helpers/expectBNIsWithinDelta');
+
+
+const TimeFrameMock = artifacts.require('TimeFrameMock');
+
+
+
+contract('TimeFrame', () => {
+  beforeEach(async () => {
+    this.now = await time.latest();
+
+    this.oneWeek = await time.duration.weeks(1);
+    this.twoWeeksAgo = await this.now - time.duration.weeks(2);
+    this.oneWeekAgo = this.now - this.oneWeek;
+    this.oneWeekFromNow = this.now + this.oneWeek;
+    this.twoWeeksFromNow = await this.now + time.duration.weeks(2);
+  });
+
+  describe('#hasStarted', () => {
+    it('it returns true when now if greater than start date', async () => {
```

Comment:
I'd personally prefer it if these tests were written in a way that they reflected how the return value of `TimeFrame` functions changes as time passes (e.g. using `time.increaseTo`), something like:

```
context before start
  it returns false

  context after start
    beforeEach
      time.advanceToStart

    it returns true
```

But that's just an idea, I won't block this PR on that :)

Reply:
I love this idea but I have a suggestion on top of that 
how about I add another test and this test remains as basic functionalities and second test we call it something like advanceTimeFrame or TimeFrameInAction and there I write 9 test's
1.before start
2.very close to starting (1 millisecond before the start)
3.equal to start
4. A little bit after the start (1 millisecond after the start)
5. middle of the time frame
6.very close to the end (1 millisecond before the end)
7.equal to end
8. A little bit after the end (1 millisecond after the end)
9. after the end

let me know your thought's on that

Reply:
Those seem great (maybe 1 millisecond is a bit overkill :sweat_smile:), but why would you also keep the basic tests? You'd be testing the exact same functionality on both.

Reply:
whatever you want boss . I make it happen ASAP